packages/frontity/src/utils/slot-and-fill/slot.tsx

```diff
@@ -28,7 +28,15 @@ type SlotType = React.FC<Connect<Package, SlotProps>>;
 /**
  * A React component that renders a Fill to fulfill a Slot and Fill pattern.
  */
-const Slot: SlotType = ({ name, state, children, ...slotProps }) => {
+const Slot: SlotType = ({
+  name,
+  state,
+  children,
+  libraries, //  These props are unused on purpose.
+  actions, ////  We're only destructuring them so that we can filter them out from `...slotProps`
+  roots, //////  which are passed to the Fill component
```

Comment:
Wow, this line makes me think that maybe `connect` is wrong:

Here, the types for `Connect`. The `name`, `roots` and `fills` props are removed:

https://github.com/frontity/frontity/blob/84cd8aeaa33bb8a34a8de499e7476eac80bb544f/packages/types/src/connect.ts#L5-L11

But here in `connect` all frontity props are passed, including `roots` and `fills`:

https://github.com/frontity/frontity/blob/2e9d6137c5d9c5e6cd5aac6b3b06bd600fba5b80/packages/connect/src/connect.js#L84-L85

https://github.com/frontity/frontity/blob/2e9d6137c5d9c5e6cd5aac6b3b06bd600fba5b80/packages/connect/src/connect.js#L116-L117

What should be the correct behavior for connect? Not to pass those props? Or add them to the `Connect` type and update all docs?

cc: @michalczaplinski @luisherranz

Reply:
To me it seems that we should pass all those props: `state`, `actions` , `name`, `roots`. So, I'm not sure why the types are erased in `Connect`@luisherranz 

Also I think that `fills` was removed, in favour or `state.fills`, correct?

Reply:
You're right that roots should not be passed down to connected components and fills is deprecated. I want to review the types once we migrate to RES.

But for this case, please use `useConnect`: https://github.com/frontity/frontity/pull/475#discussion_r440256221